apkid/rules/elf/obfuscators.yara

```diff
@@ -155,6 +155,25 @@ rule ollvm : obfuscator
     not ollvm_v6_0_strenc
 }
 
+rule byteguard : obfuscator
+{
+  meta:
+    description = "ByteGuard 0.9"
+    sample      = "eed4f7b907fe2173935d307dfb0d6aa7098f69db8dfb65e49affd7b7a6c0a5e4"
+    samples     = "https://koodous.com/rulesets/5862/apks"
+    author      = "Eduardo Novella"
+
+  strings:
+    // clang version 6.0.0 (Byteguard 0.6) (git@sysrepo.byted.org:dingbaozeng/native_obfuscator.git 448f20ff6eb06dd336dd81846d6a7dc8ba8c961b)
+    // Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
+    $clang_version = "clang version 6.0.0 (Byteguard 0.6) (git@sysrepo.byted.org:dingbaozeng/native_obfuscator.git "
+    $based_on      = "Apple LLVM version 6.0.0 (ByteGuard 0.9."
+    $joke          = " -_-||| Sorry! You can't touch me any longer ... "
```

Comment:
Is this for byteguard 0.9 only?
Or does this detect both byteguard 0.6 and 0.9?

The strings are confusing in here since there are two specific versions tagged to one description w/ a version number, but the condition is for at least 2.

If this detects two seperate versions I'd suggested;
1. detect version 0.6
2. detect version 0.9
3. generic rule which triggers regardless of version number assuming 0.6 and 0.9 where not detected.

Reply:
No idea about versioning. Also realized about the mismatch in the versions

Reply:
Only two versions so far
```sh
[01:51 edu@xps byteguard] >  strings -f 69810b5764be24ea35604e5b4a7ee1fe1b5566e7f7e4fb554fa05e73a1781d78.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
69810b5764be24ea35604e5b4a7ee1fe1b5566e7f7e4fb554fa05e73a1781d78.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 69810b5764be24ea35604e5b4a7ee1fe1b5566e7f7e4fb554fa05e73a1781d78.apk.unpack/lib/armeabi/libnms.so|egrep -i byteguard
69810b5764be24ea35604e5b4a7ee1fe1b5566e7f7e4fb554fa05e73a1781d78.apk.unpack/lib/armeabi/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 69810b5764be24ea35604e5b4a7ee1fe1b5566e7f7e4fb554fa05e73a1781d78.apk.unpack/lib/arm64-v8a/libnms.so|egrep -i byteguard
69810b5764be24ea35604e5b4a7ee1fe1b5566e7f7e4fb554fa05e73a1781d78.apk.unpack/lib/arm64-v8a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 69810b5764be24ea35604e5b4a7ee1fe1b5566e7f7e4fb554fa05e73a1781d78.apk.unpack/lib/x86/libnms.so|egrep -i byteguard
69810b5764be24ea35604e5b4a7ee1fe1b5566e7f7e4fb554fa05e73a1781d78.apk.unpack/lib/x86/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 085a7a0b1d412c316be178376fbcf8c1a1f576475e7ca7e8489cdb69cdb08be2.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
085a7a0b1d412c316be178376fbcf8c1a1f576475e7ca7e8489cdb69cdb08be2.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 085a7a0b1d412c316be178376fbcf8c1a1f576475e7ca7e8489cdb69cdb08be2.apk.unpack/lib/armeabi/libnms.so|egrep -i byteguard
085a7a0b1d412c316be178376fbcf8c1a1f576475e7ca7e8489cdb69cdb08be2.apk.unpack/lib/armeabi/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 085a7a0b1d412c316be178376fbcf8c1a1f576475e7ca7e8489cdb69cdb08be2.apk.unpack/lib/arm64-v8a/libnms.so|egrep -i byteguard
085a7a0b1d412c316be178376fbcf8c1a1f576475e7ca7e8489cdb69cdb08be2.apk.unpack/lib/arm64-v8a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f aff37c3b4efb0616ffd2e878958199df11e63563bc230ff2024d9f9a873a2bbb.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
aff37c3b4efb0616ffd2e878958199df11e63563bc230ff2024d9f9a873a2bbb.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f c3cd34fd61d504fffb22fa7c79dcd677b03572fad63a5f7ddcf25b7b452c78fd.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
c3cd34fd61d504fffb22fa7c79dcd677b03572fad63a5f7ddcf25b7b452c78fd.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f b4ed3f24c969d2cea22efd4a5b8fde8f79264e34472fe9c90550158f9036c609.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
b4ed3f24c969d2cea22efd4a5b8fde8f79264e34472fe9c90550158f9036c609.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 15fcd263136e735b3c3d826ff3811df5aa491710de4911320a794549d7327f2d.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
15fcd263136e735b3c3d826ff3811df5aa491710de4911320a794549d7327f2d.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 13b0695addc8f1cdc6aaa3039a6102f6d3a50de478030a6831c0a4602dec942a.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
13b0695addc8f1cdc6aaa3039a6102f6d3a50de478030a6831c0a4602dec942a.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f b8a36fc9404bf8a326c109a8c1cc0c27cee63f504953bd1e316a49a2e2d1daa8.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
b8a36fc9404bf8a326c109a8c1cc0c27cee63f504953bd1e316a49a2e2d1daa8.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 04eb28a86f1215d59f9544e48953e16252695cc09048767a5bc2729c08254f0d.apk.unpack/lib/x86_64/libnms.so|egrep -i byteguard
04eb28a86f1215d59f9544e48953e16252695cc09048767a5bc2729c08254f0d.apk.unpack/lib/x86_64/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 04eb28a86f1215d59f9544e48953e16252695cc09048767a5bc2729c08254f0d.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
04eb28a86f1215d59f9544e48953e16252695cc09048767a5bc2729c08254f0d.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 04eb28a86f1215d59f9544e48953e16252695cc09048767a5bc2729c08254f0d.apk.unpack/lib/arm64-v8a/libnms.so|egrep -i byteguard
04eb28a86f1215d59f9544e48953e16252695cc09048767a5bc2729c08254f0d.apk.unpack/lib/arm64-v8a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 04eb28a86f1215d59f9544e48953e16252695cc09048767a5bc2729c08254f0d.apk.unpack/lib/x86/libnms.so|egrep -i byteguard
04eb28a86f1215d59f9544e48953e16252695cc09048767a5bc2729c08254f0d.apk.unpack/lib/x86/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f a8927291c121927557a01d5712de9bd004355f5a41646ddab90d8b002d39bcff.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
a8927291c121927557a01d5712de9bd004355f5a41646ddab90d8b002d39bcff.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f a8927291c121927557a01d5712de9bd004355f5a41646ddab90d8b002d39bcff.apk.unpack/lib/arm64-v8a/libnms.so|egrep -i byteguard
a8927291c121927557a01d5712de9bd004355f5a41646ddab90d8b002d39bcff.apk.unpack/lib/arm64-v8a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f a8927291c121927557a01d5712de9bd004355f5a41646ddab90d8b002d39bcff.apk.unpack/lib/x86/libnms.so|egrep -i byteguard
a8927291c121927557a01d5712de9bd004355f5a41646ddab90d8b002d39bcff.apk.unpack/lib/x86/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 0034a452fa3f0bb4a4f11a0dd720a54e25ffab7815e1ebdc0ec9307de1d93e84.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
0034a452fa3f0bb4a4f11a0dd720a54e25ffab7815e1ebdc0ec9307de1d93e84.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 0034a452fa3f0bb4a4f11a0dd720a54e25ffab7815e1ebdc0ec9307de1d93e84.apk.unpack/lib/armeabi/libnms.so|egrep -i byteguard
0034a452fa3f0bb4a4f11a0dd720a54e25ffab7815e1ebdc0ec9307de1d93e84.apk.unpack/lib/armeabi/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 0034a452fa3f0bb4a4f11a0dd720a54e25ffab7815e1ebdc0ec9307de1d93e84.apk.unpack/lib/arm64-v8a/libnms.so|egrep -i byteguard
0034a452fa3f0bb4a4f11a0dd720a54e25ffab7815e1ebdc0ec9307de1d93e84.apk.unpack/lib/arm64-v8a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f b46a1dc51352e50068287a448e288d20d36a5e0ba3a0145cdb50e0c7e45c3f9a.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
b46a1dc51352e50068287a448e288d20d36a5e0ba3a0145cdb50e0c7e45c3f9a.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 1bb3aca29e034712651680f2013c05ebfae9cca20ec9f6b6e02e76d482d7aa62.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
1bb3aca29e034712651680f2013c05ebfae9cca20ec9f6b6e02e76d482d7aa62.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 71181bd1692233a938d2ef45b0911d8227696202293b05c2cb568b73ffeba59d.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
71181bd1692233a938d2ef45b0911d8227696202293b05c2cb568b73ffeba59d.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 71181bd1692233a938d2ef45b0911d8227696202293b05c2cb568b73ffeba59d.apk.unpack/lib/armeabi/libnms.so|egrep -i byteguard
71181bd1692233a938d2ef45b0911d8227696202293b05c2cb568b73ffeba59d.apk.unpack/lib/armeabi/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 71181bd1692233a938d2ef45b0911d8227696202293b05c2cb568b73ffeba59d.apk.unpack/lib/arm64-v8a/libnms.so|egrep -i byteguard
71181bd1692233a938d2ef45b0911d8227696202293b05c2cb568b73ffeba59d.apk.unpack/lib/arm64-v8a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 178b1ef3c4ac563604c8a262f0e3651f56995768c8aa13ccc845f33bd6eb0ac2.apk.unpack/lib/armeabi-v7a/libcjtfcc.so|egrep -i byteguard
178b1ef3c4ac563604c8a262f0e3651f56995768c8aa13ccc845f33bd6eb0ac2.apk.unpack/lib/armeabi-v7a/libcjtfcc.so: clang version 5.0.2 (Byteguard-0.9.2-255c7b5e)
[01:51 edu@xps byteguard] >  strings -f 178b1ef3c4ac563604c8a262f0e3651f56995768c8aa13ccc845f33bd6eb0ac2.apk.unpack/lib/armeabi-v7a/libcms.so|egrep -i byteguard
178b1ef3c4ac563604c8a262f0e3651f56995768c8aa13ccc845f33bd6eb0ac2.apk.unpack/lib/armeabi-v7a/libcms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f afbe49cb2e8d97a75882ff19c81f8e9e6a1673f4ba88d24936c51af25155c408.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
afbe49cb2e8d97a75882ff19c81f8e9e6a1673f4ba88d24936c51af25155c408.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f afbe49cb2e8d97a75882ff19c81f8e9e6a1673f4ba88d24936c51af25155c408.apk.unpack/lib/x86/libnms.so|egrep -i byteguard
afbe49cb2e8d97a75882ff19c81f8e9e6a1673f4ba88d24936c51af25155c408.apk.unpack/lib/x86/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 1f2fac9229c81533f59b9ebc8b73c63d2a1736055ee2a95622d4d463290e5b81.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
1f2fac9229c81533f59b9ebc8b73c63d2a1736055ee2a95622d4d463290e5b81.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 1f2fac9229c81533f59b9ebc8b73c63d2a1736055ee2a95622d4d463290e5b81.apk.unpack/lib/armeabi/libnms.so|egrep -i byteguard
1f2fac9229c81533f59b9ebc8b73c63d2a1736055ee2a95622d4d463290e5b81.apk.unpack/lib/armeabi/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f b94eb0a640b8e3af0167e1d1f2fada5c293fe405f6543c9ee9cc03922cae288a.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
b94eb0a640b8e3af0167e1d1f2fada5c293fe405f6543c9ee9cc03922cae288a.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f f8fe7eb58a3149e7416386e0c9b371312426dde6e40aa918ef74aa6be6c6d8d9.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
f8fe7eb58a3149e7416386e0c9b371312426dde6e40aa918ef74aa6be6c6d8d9.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f f8fe7eb58a3149e7416386e0c9b371312426dde6e40aa918ef74aa6be6c6d8d9.apk.unpack/lib/armeabi/libnms.so|egrep -i byteguard
f8fe7eb58a3149e7416386e0c9b371312426dde6e40aa918ef74aa6be6c6d8d9.apk.unpack/lib/armeabi/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f f8fe7eb58a3149e7416386e0c9b371312426dde6e40aa918ef74aa6be6c6d8d9.apk.unpack/lib/arm64-v8a/libnms.so|egrep -i byteguard
f8fe7eb58a3149e7416386e0c9b371312426dde6e40aa918ef74aa6be6c6d8d9.apk.unpack/lib/arm64-v8a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f ed0c6f3ee0aaa55188ccc0e6aefce6eb7095d3f3974b38c8e6a577e90e9da536.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
ed0c6f3ee0aaa55188ccc0e6aefce6eb7095d3f3974b38c8e6a577e90e9da536.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f a6c989e94a043e5d003063b63c5cc4a85fae4bf4d44e3ee0a9195b50c5b864a4.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
a6c989e94a043e5d003063b63c5cc4a85fae4bf4d44e3ee0a9195b50c5b864a4.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/x86_64/libnms.so|egrep -i byteguard
bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/x86_64/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/armeabi/libnms.so|egrep -i byteguard
bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/armeabi/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/arm64-v8a/libnms.so|egrep -i byteguard
bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/arm64-v8a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/x86/libnms.so|egrep -i byteguard
bef8a24db3ed18ff0b4d59a33adee4789ee0a8b2fc835bbb0d157bd0b8d7a298.apk.unpack/lib/x86/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 7961dfc96c7a81f8f1d818dc891813ef64770b31ac4bd9712585d9eb387a564a.apk.unpack/lib/armeabi/libnms.so|egrep -i byteguard
7961dfc96c7a81f8f1d818dc891813ef64770b31ac4bd9712585d9eb387a564a.apk.unpack/lib/armeabi/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 4c7f63be3c06607adbdf562877bea4fb67714b3c4981a46aec00508d278c9537.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
4c7f63be3c06607adbdf562877bea4fb67714b3c4981a46aec00508d278c9537.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
[01:51 edu@xps byteguard] >  strings -f 36fbee43fb87984528cbc3f2d95218567f991533bf10d6f8654b6b8946b974a9.apk.unpack/lib/armeabi-v7a/libnms.so|egrep -i byteguard
36fbee43fb87984528cbc3f2d95218567f991533bf10d6f8654b6b8946b974a9.apk.unpack/lib/armeabi-v7a/libnms.so: Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
```

Reply:
I don't follow, so do all those samples only hit on 1 string?

Do we need the other two?

Reply:
I've tackled the versioning. Ready to merge